entry.c

```diff
@@ -530,10 +538,16 @@ extern void closeTagFile (const boolean resize)
 
 	if (Option.etags)
 		writeEtagsIncludes (TagFile.fp);
+	abort_if_ferror (TagFile.fp);
 	desiredSize = ftell (TagFile.fp);
 	fseek (TagFile.fp, 0L, SEEK_END);
 	size = ftell (TagFile.fp);
-	fclose (TagFile.fp);
+	if (fclose (TagFile.fp) != 0)
+	{
+		error (FATAL | PERROR, "cannot close tag file");
+		exit (1);
```

Comment:
`error(FATAL)` will `exit` itself, so the `exit(1)` is dead code.  And if didn't, `abort_if_ferror()` should also `exit()` then.